packages/xdl/src/project/getPublishExpConfigAsync.ts

```diff
@@ -30,22 +31,27 @@ export async function getPublishExpConfigAsync(
   options.releaseChannel = options.releaseChannel || 'default';
 
   // Verify that exp/app.json and package.json exist
-  const { exp: privateExp } = getConfig(projectRoot);
-  const { hooks } = privateExp;
-  const { exp, pkg } = getConfig(projectRoot, { isPublicConfig: true });
+  const {
+    exp: { hooks, runtimeVersion },
+  } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+  const { exp, pkg } = getConfig(projectRoot, {
+    isPublicConfig: true,
+    skipSDKVersionRequirement: !!runtimeVersion,
+  });
 
-  const { sdkVersion } = exp;
+  // we can't publish with both sdkVersion and runtimeVerson specified, so ensure we pick one or the other.
+  const xorExp = runtimeVersion
+    ? ({ ...exp, sdkVersion: undefined } as ExpoAppManifestWithRuntimeVersion)
+    : (exp as ExpoAppManifestWithSdk);
```

Comment:
Maybe it'd be worth writing a simple unit test on `getPublishExpConfigAsync` which provides two different types of configs (runtimeVersion, sdkVersion) and gets the expected results.